sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/GroupIntoBatches.java

```diff
@@ -66,25 +68,49 @@
  *         }
  *     }}));
  * </pre>
+ *
+ * <p>Example 2 (batch unbounded input in a global window):
+ *
+ * <pre>{@code
+ * PCollection<KV<String, String>> unboundedInput = ...;
+ * long batchSize = 100L;
+ * Duration maxBufferingDuration = Duration.standardSeconds(10);
+ * PCollection<KV<String, Iterable<String>>> batched = unboundedInput
+ *     .apply(Window.<KV<String, String>>into(new GlobalWindows())
+ *         .triggering(Repeatedly.forever(AfterPane.elementCountAtLeast(1)))
+ *         .discardingFiredPanes())
+ *     .apply(GroupIntoBatches.<String, String>ofSize(batchSize)
+ *         .withMaxBufferingDuration(maxBufferingDuration));
+ * }</pre>
  */
 public class GroupIntoBatches<K, InputT>
     extends PTransform<PCollection<KV<K, InputT>>, PCollection<KV<K, Iterable<InputT>>>> {
 
   private final long batchSize;
+  private final Duration maxBufferingDuration;
 
-  private GroupIntoBatches(long batchSize) {
+  private GroupIntoBatches(long batchSize, Duration maxBufferingDuration) {
     this.batchSize = batchSize;
+    this.maxBufferingDuration = maxBufferingDuration;
   }
 
   public static <K, InputT> GroupIntoBatches<K, InputT> ofSize(long batchSize) {
-    return new GroupIntoBatches<>(batchSize);
+    return new GroupIntoBatches<>(batchSize, Duration.ZERO);
   }
 
   /** Returns the size of the batch. */
   public long getBatchSize() {
     return batchSize;
   }
 
+  /**
+   * Set a time limit (in processing time) on how long an incomplete batch of elements is allowed to
+   * be buffered. Once a batch is flushed to output, the timer is reset.
+   */
+  public GroupIntoBatches<K, InputT> withMaxBufferingDuration(Duration duration) {
+    return new GroupIntoBatches<>(batchSize, duration);
```

Comment:
If `Duration.ZERO` is not preferred, we can have a check here.

Reply:
Added a check. But we still need additional checks on null value. See my reply below.